diff --git a/dist/index.cjs b/dist/index.cjs
index a2f92474c489df2ba29162fa1538ebfecefc9e5e..c4a1a61c0965f6b5487292abbf3ec07ddcf84db6 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -69,7 +69,7 @@ function getWorkspaceArgs(options) {
   }
   const workspacePkg = typeof options.workspace === "string" && options.workspace !== "" ? options.workspace : void 0;
   if (options.packageManager.name === "pnpm") {
-    return workspacePkg ? ["--dir", workspacePkg] : ["--workspace-root"];
+    return workspacePkg ? ["--filter", workspacePkg] : ["--workspace-root"];
   }
   if (options.packageManager.name === "npm") {
     return workspacePkg ? ["-w", workspacePkg] : ["--workspaces"];
diff --git a/dist/index.mjs b/dist/index.mjs
index 8ed37f3cd879bbac536799d881eefb42b6440c25..a293599d56e8e19fd4f821bc1a43b7a0fbcd69f2 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -67,7 +67,7 @@ function getWorkspaceArgs(options) {
   }
   const workspacePkg = typeof options.workspace === "string" && options.workspace !== "" ? options.workspace : void 0;
   if (options.packageManager.name === "pnpm") {
-    return workspacePkg ? ["--dir", workspacePkg] : ["--workspace-root"];
+    return workspacePkg ? ["--filter", workspacePkg] : ["--workspace-root"];
   }
   if (options.packageManager.name === "npm") {
     return workspacePkg ? ["-w", workspacePkg] : ["--workspaces"];
